### PR TITLE
Add Hugging Face Transformers postprocessing integration

### PR DESCRIPTION
**Pull request by Leev1s**
## Summary

This PR adds an official Hugging Face `transformers` integration path to RankSEG for semantic segmentation workflows.

The main goal is to let users keep the standard `processor -> model -> outputs` inference pattern and insert RankSEG at the post-processing boundary.

### 1. `rankseg.transformers` integration helper

This PR adds a new compatibility layer at:

- `rankseg.transformers.postprocess(...)`
- `rankseg.transformers.restore_semantic_probs(...)`

The helper restores supported Hugging Face segmentation outputs into semantic probability maps at the original image size, then applies RankSEG as the final post-processing step.

For a runnable demonstration, the README links to this [Colab notebook](https://colab.research.google.com/github/Leev1s/rankseg/blob/feat/transformers-adapter/notebooks/rankseg_with_transformers.ipynb), which shows the Transformers integration end to end.

Supported output families in this PR:

- `outputs.logits`
- `outputs.class_queries_logits + outputs.masks_queries_logits`
- `outputs.logits + outputs.pred_masks`
- `outputs.semantic_seg`

The implementation is intentionally narrow and follows the official `transformers` semantic post-processing behavior for the supported families.

### 2. Official example and notebook

This PR adds:

- `examples/transformers_rankseg.py`
- `notebooks/rankseg_with_transformers.ipynb`

The example script shows the minimal insertion point in a standard Hugging Face inference flow.

The notebook is written as a side-by-side baseline vs. RankSEG integration walkthrough, with the integration hand-off kept explicitly at `outputs = model(**inputs)`.

### 3. Documentation updates

This PR updates the repository documentation so that `Transformers` is now presented as an official integration path alongside `PyTorch Native`.

Changes include:

- README and README_zh updates
- a new `doc/source/integrations_transformers.rst` page
- docs navigation updates in `index.rst`, `getting_started.rst`, and `integrations.rst`
- API docs entries for `rankseg.transformers.postprocess` and `rankseg.transformers.restore_semantic_probs`

### 4. Test coverage

This PR adds `tests/test_transformers.py` to cover:

- supported output restoration paths
- DETR / Conditional DETR branching
- Mask2Former and SAM3 behavior alignment
- explicit failure cases for unsupported output structures
- default `postprocess(...)` behavior

## Validation

I validated this work locally with:

- `uv run --with pytest --with torchmetrics --with scipy python -m pytest -o addopts='' tests/test_transformers.py`
- `uv run --with-requirements doc/requirements.txt sphinx-build -b html doc/source /tmp/rankseg-docs`

## Note

A small number of notebook and Colab links currently point to my fork to keep the demo runnable during PR review. These can be updated to the upstream repository after merge.

### CODE REVIEW

**Comment by statmlben** [comment]
Thank you @Leev1s 

Main items to fix before merge:

1. Add integrations_transformers to doc/source/integrations.rst:11. Sphinx currently warns that doc/source/integrations_transformers.rst is not included in any toctree.
3. Replace contributor-branch links before merge:
      - README.md:77
      - README_zh.md:78
      - doc/source/integrations_transformers.rst
      - notebooks/rankseg_with_transformers.ipynb

These currently point to Leev1s/rankseg / feat/transformers-adapter. After merge they should point to rankseg/rankseg on main, or use the normal PyPI install path.

**Comment by Leev1s** [comment]
Thank you @statmlben for the guidance! I have updated the PR before merge:

- Added `integrations_transformers` to `doc/source/integrations.rst` so the Transformers docs page is included in the Sphinx toctree.
- Replaced the contributor-branch Colab links in `README.md`, `README_zh.md`, and `doc/source/integrations_transformers.rst` with the upstream `rankseg/rankseg` main link.
- Updated the notebook setup to use the normal PyPI install path: `uv pip install -U rankseg`.

I also rechecked the PR diff and confirmed the Transformers-related `Leev1s/rankseg` / `feat/transformers-adapter` links are gone.
